### PR TITLE
docs: retested and modernized all the docstrings in highlevel.py.

### PR DESCRIPTION
**Pull request by jpivarski**
This finishes the work started in #1946, commit dc0f91e1978ee3f364ed4b3c9abb5181798be99b.

All of the sample code in docstrings is being tested and modernized: `print` statements are being removed in favor of `show` and the output is being replaced with v2 output.

----
:books: The documentation for this PR will be available at <https://awkward-array.readthedocs.io/en/jpivarski-modernize-highlevel-docstrings/> once Read the Docs has finished building :hammer:

### CODE REVIEW

**Comment by jpivarski** [comment]
100% of these changes are in code snippets in docstrings.

File: src/awkward/operations/ak_combinations.py [hunk 1]
```diff
@@ -48,21 +48,21 @@ def combinations(
...
-    As a simple example with `axis=0`, consider the following `array`
+    As a simple example with `axis=0`, consider the following
 
-        ak.Array(["a", "b", "c", "d", "e"])
+        >>> array = ak.Array(["a", "b", "c", "d", "e"])
 
     The combinations choose `2` are:
 
-        >>> ak.to_list(ak.combinations(array, 2, axis=0))
+        >>> ak.combinations(array, 2, axis=0).show()
```
Comment (agoose77) [inline src/awkward/operations/ak_combinations.py:57]: This was such a big QoL improvement.

File: src/awkward/operations/ak_count.py [hunk 2]
```diff
@@ -35,13 +35,13 @@ def count(
...
-        ak.Array([[ 0.1,  0.2      ],
-                  [None, 10.2, None],
-                  None,
-                  [20.1, 20.2, 20.3],
-                  [30.1, 30.2      ]])
+        >>> array = ak.Array([[ 0.1,  0.2      ],
+        ...                   [None, 10.2, None],
+        ...                    None,
+        ...                    [20.1, 20.2, 20.3],
+        ...                    [30.1, 30.2      ]])
```
Comment (agoose77) [inline src/awkward/operations/ak_count.py]: Remove a leading space?
```suggestion
        ...                   [None, 10.2, None],
        ...                   None,
        ...                   [20.1, 20.2, 20.3],
        ...                   [30.1, 30.2      ]])
```

File: src/awkward/operations/ak_to_categorical.py [hunk 3]
```diff
@@ -30,17 +30,17 @@ def to_categorical(array, *, highlevel=True, behavior=None):
         >>> array = ak.Array([["one", "two", "three"], [], ["three", "two"]])
         >>> categorical = ak.to_categorical(array)
         >>> categorical
-        <Array [['one', 'two', ... 'three', 'two']] type='3 * var * categorical[type=str...'>
-        >>> ak.type(categorical)
+        <Array [['one', 'two', 'three'], ..., [...]] type='3 * var * categorical[ty...'>
+        >>> categorical.type.show()
         3 * var * categorical[type=string]
-        >>> ak.to_list(categorical) == ak.to_list(array)
+        >>> categorical.tolist() == array.tolist()
```
Comment (agoose77) [inline src/awkward/operations/ak_to_categorical.py]: Should we promote `to_list` over `tolist`?
```suggestion
        >>> categorical.to_list() == array.to_list()
```

**Review by agoose77 — CHANGES_REQUESTED** [review]

**Comment by agoose77** [comment]
You have the go-ahead to merge once you've either changed or ignored the discussion points! :)

File: src/awkward/operations/ak_count.py (hunk 2 above)
Comment (jpivarski) [reply src/awkward/operations/ak_count.py]: Yes, that's right: semantically, the spacing should be one to the left.

File: src/awkward/operations/ak_to_categorical.py [hunk 4]
```diff
@@ -55,16 +55,19 @@ def to_categorical(array, *, highlevel=True, behavior=None):
...
         <Array [{x: 1.1, y: 'one'}, ... y: 'one'}] type='5 * {"x": categorical[type=floa...'>
-        >>> ak.type(categorical_records)
-        5 * {"x": categorical[type=float64], "y": categorical[type=string]}
-        >>> ak.to_list(categorical_records) == ak.to_list(records)
+        >>> categorical_records.type.show()
+        5 * {
+            x: categorical[type=float64],
+            y: categorical[type=string]
+        }
+        >>> categorical_records.tolist() == records.tolist()
```
Comment (jpivarski) [inline src/awkward/operations/ak_to_categorical.py]: If that's what we're doing, here's another one.

```suggestion
        >>> categorical_records.to_list() == records.to_list()
```

**Comment by jpivarski** [comment]
@agoose77 I've enabled auto-merge, and only re-requested the review so that you can make it not red. It won't stop the merge.

Thanks!

**Review by agoose77 — APPROVED** [review]

**Pull request merged**